apps/anoma_node/lib/supervisor.ex

```diff
@@ -75,9 +75,25 @@ defmodule Anoma.Supervisor do
   """
   @spec stop_node(String.t()) :: :ok
   def stop_node(node_id) do
-    Anoma.Node.Registry.via(node_id, Anoma.Node.Supervisor)
-
-    Supervisor.stop(Anoma.Node.Registry.via(node_id, Anoma.Node.Supervisor))
+    via = Anoma.Node.Registry.via(node_id, Anoma.Node.Supervisor)
+
+    case GenServer.whereis(via) do
+      nil ->
+        # Already stopped / never started
+        :ok
+
+      pid when is_pid(pid) ->
+        # Prefer DynamicSupervisor API so the child table is updated atomically
+        case DynamicSupervisor.terminate_child(Anoma.Node.NodeSupervisor, pid) do
+          :ok ->
+            :ok
+
+          # If it's no longer tracked by the DS (race), stop directly and treat as success
+          {:error, :not_found} ->
+            _ = Supervisor.stop(pid)
+            :ok
+        end
+    end
```

Comment:
This doesn't make a functional difference.

```elixir
iex(anoma@YU-NO)16> node = Anoma.Node.Examples.ENode.start_node
%Anoma.Node.Examples.ENode{
  pid: #PID<0.982.0>,
  node_id: "D6357773AE9666F46E5ECC81EB3D7ED1907B4ECA30209495DF1C251BEF13893B"
}
iex(anoma@YU-NO)17> GenServer.whereis(Anoma.Node.Registry.via(node.node_id, Anoma.Node.Supervisor))
#PID<0.982.0>
iex(anoma@YU-NO)18> Anoma.Supervisor.stop_node(node.node_id)
:ok
iex(anoma@YU-NO)19> GenServer.whereis(Anoma.Node.Registry.via(node.node_id, Anoma.Node.Supervisor))
nil
```

without the patch included.

If one is trying to stop a node that doesn't exist it's fine for it to do nothing